From dc2270c16ffcf320d66787852bd0fcd010a7c1e4 Mon Sep 17 00:00:00 2001
From: "Bhakte, GurudattaX" <gurudattax.bhakte@intel.com>
Date: Fri, 13 Mar 2015 16:45:16 +0530
Subject: [PATCH] ASoC: Intel: Byt: Fix Klockwork error in Baytrail Machine
 Driver.

Issue: GMINL-7442

Kw error: Pointer 'byt_get_codec(card)' returned from call to function
'byt_get_codec', may be NULL and will be dereferenced.
This patch fixes this issue by avoiding invocation of function in
argument and uses NULL pointer check.

Change-Id: If6c98d48180d58bc70fb21cafdb6fa099f34a805
Signed-off-by: Bhakte, GurudattaX <gurudattax.bhakte@intel.com>
---
 sound/soc/intel/board/byt_cr_dpcm_rt5640.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
index 4355d061d9eb..761e80649b3e 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
@@ -813,7 +813,7 @@ static int byt_init(struct snd_soc_pcm_runtime *runtime)
 	/* Other jack/bp stuff */
 
 	/* JACK_DET_N signal as JD-source */
-	snd_soc_update_bits(byt_get_codec(card), RT5640_JD_CTRL,
+	snd_soc_update_bits(codec, RT5640_JD_CTRL,
 			RT5640_JD_MASK, RT5640_JD_JD1_IN4P);
 
 	/* Prevent sta_jd_internal to trigger IRQ in CODEC-mode */
-- 
2.1.0

